src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -954,7 +954,17 @@ internal bool VerifyRemoteCertificate(RemoteCertificateValidationCallback? remot
 
             try
             {
-                _remoteCertificate = CertificateValidationPal.GetRemoteCertificate(_securityContext, out remoteCertificateStore);
+                X509Certificate2? remoteCertificate = CertificateValidationPal.GetRemoteCertificate(_securityContext, out remoteCertificateStore);
+
+                if (remoteCertificate != null && _remoteCertificate != null &&
+                    remoteCertificate.Equals(_remoteCertificate))
```

Comment:
This equality is, sadly, pretty weak.  You'll need to use `remoteCertificate.RawData.SequenceEquals(_remoteCertificate.RawData)`

Reply:
ok. I can do that. Would Thumbprint be any better @bartonjs? With Exception of real renegotiation, this should really not happen outside of Windows and TLS13 as the _remoteCertificate would be NULL in other cases.

Reply:
> Would Thumbprint be any better @bartonjs?

Nope, it has different problems.  Both Equals and Thumbprint can be used to determine that it's not a match, but only RawData comparison can be used to determine that it is a match.

Reply:
So Equals doesn't actually validate equality? Shouldn't we, umm, fix that?

Reply:
It is [documented](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate.equals?view=net-5.0#System_Security_Cryptography_X509Certificates_X509Certificate_Equals_System_Security_Cryptography_X509Certificates_X509Certificate_) @stephentoub that it compares serial number and issuer. (not arguing that is right)

Reply:
That also says to use Thumbprint, which Jeremy just said not to do. So my stock in the docs here is limited :-)

Reply:
Yeah, that was written before SHA-1 was broken.  (Thumbprint is the SHA-1 of the RawData, and SHA-1 is no longer allowed to be used for equality)

Reply:
updated. I'll follow-up separately to update the docs.